pinot-server/src/main/java/org/apache/pinot/server/starter/helix/BaseServerStarter.java

```diff
@@ -1058,4 +1072,19 @@ private void initSegmentFetcher(PinotConfiguration config)
   protected AdminApiApplication createServerAdminApp() {
     return new AdminApiApplication(_serverInstance, _accessControlFactory, _serverConf);
   }
+
+  private void refreshMessageCount() {
+    try {
+      HelixDataAccessor dataAccessor = _helixManager.getHelixDataAccessor();
+      List<String> children = dataAccessor.getBaseDataAccessor()
+          .getChildNames(String.format("/%s/INSTANCES/%s/MESSAGES", _helixClusterName, _instanceId), 0);
+      _cachedMessageCount.set(children == null ? 0 : children.size());
+    } catch (Exception e) {
+      LOGGER.warn("Failed to get Helix message count", e);
+      _cachedMessageCount.set(0);
```

Comment:
just thinking out loud, should we set this to a -ve value on error? okay with leaving as 0, but this might be misleading

Reply:
yeah makes sense, let's retain the value